src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -185,26 +171,34 @@ public InternalEngine(EngineConfig engineConfig, boolean skipInitialTranslogReco
         logger.trace("created new InternalEngine");
     }
 
+    private Translog openTranslog(EngineConfig engineConfig, IndexWriter writer, boolean createNew) throws IOException {
+        final Translog.TranslogGeneration generation = loadTranslogIdFromCommit(writer);
+        final TranslogConfig translogConfig = engineConfig.getTranslogConfig();
+        if (createNew == false) {
+            translogConfig.setTranslogGeneration(generation);
+            if (generation != null && generation.translogUUID == null) {
+                // only upgrade on pre-2.0 indices...
+                Translog.upgradeLegacyTranslog(logger, translogConfig);
+            }
+        }
+        final Translog translog = new Translog(translogConfig);
+        if (generation == null) {
+            if (createNew) {
```

Comment:
yeah it has a weird meaning. I pushed a new commit to clarify and changed the condition to `(createNew == false)`